filament/backend/src/opengl/OpenGLDriver.cpp

```diff
@@ -3112,7 +3145,17 @@ void OpenGLDriver::draw(PipelineState state, Handle<HwRenderPrimitive> rph) {
 
     useProgram(p);
 
-    const GLRenderPrimitive* rp = handle_cast<const GLRenderPrimitive *>(rph);
+    GLRenderPrimitive* rp = handle_cast<GLRenderPrimitive *>(rph);
+
+    // If necessary, mutate the bindings in the VAO.
+    VertexBufferHandle vbwo = rp->gl.vertexBufferWithObjects;
+    if (UTILS_UNLIKELY(vbwo)) {
+        const GLVertexBuffer const* vb = handle_cast<GLVertexBuffer*>(vbwo);
+        if (vb->bufferObjectsVersion != rp->gl.vertexBufferVersion) {
```

Comment:
wouldn't it be *slightly* better to swap the sides of the `!=`, i.e. "if the rp is out of date".